src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
+                        logger.debug("Found at least one file match on %s: %s", data.hosts, line)
+                        hosts_with_files.add(data.hosts)
+                        break
+                if re.findall(fr"{FAILURE_TRIGGER}", "\n".join(data.stdout)):
+                    # If a file is found containing the FAILURE_TRIGGER then report a failure.
+                    # This feature is used by avocado tests to verify that launch.py reports
+                    # errors correctly in CI. See test_launch_failures in harness/basic.py for
+                    # more details.
+                    logger.debug(
+                        "Found a file matching the '%s' failure trigger on %s",
+                        FAILURE_TRIGGER, data.hosts)
+                    message = f"Error trigger failure file found in {source} (error handling test)"
+                    self._fail_test(self.result.tests[-1], "Process", message)
+                    hosts_with_files.add(data.hosts)
+                    status = 16
+
+        logger.debug("List files results: status=%s, hosts_with_files=%s", status, hosts_with_files)
+        return status, hosts_with_files
+
+    def _check_log_size(self, hosts, source, pattern, depth, threshold):
+        """Check if any file sizes exceed the threshold.
 
-    """
-    file_parts = os.path.split(test_file)
-    return "-".join(
-        [os.path.splitext(os.path.basename(part))[0] for part in file_parts])
+        Args:
+            hosts (NodSet): hosts on which the files are located
+            source (str): where the files are currently located
+            pattern (str): pattern used to limit which files are processed
+            depth (int): max depth for find command
+            threshold (str): optional upper size limit for test log files
 
+        Returns:
+            int: status code: 0 = success, 32 = failure
 
-def stop_daos_agent_services(test_file, args):
-    """Stop any daos_agent.service running on the hosts running servers.
+        """
+        source_files = os.path.join(source, pattern)
+        logger.debug("-" * 80)
+        logger.debug(
+            "Checking for any %s files exceeding %s on %s", source_files, threshold, hosts)
+        other = ["-size", f"+{threshold}", "-printf", "'%p %k KB'"]
+        result = run_remote(logger, hosts, find_command(source, pattern, depth, other))
+        if not result.passed:
+            message = f"Error checking for {source_files} files exceeding the {threshold} threshold"
+            self._fail_test(self.result.tests[-1], "Process", message)
+            return 32
+
+        # The command output will include the source path if the threshold has been exceeded
+        for data in result.output:
+            if source in "\n".join(data.stdout):
+                message = f"One or more {source_files} files exceeded the {threshold} threshold"
+                self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call